src/search/include/search/search.h

```diff
@@ -70,6 +70,75 @@ has_satisfiable_ata_configuration(
 	});
 }
 
+namespace details {
+
+template <typename Location, typename ActionType, typename ConstraintSymbolType>
+void
+label_graph(SearchTreeNode<Location, ActionType, ConstraintSymbolType> *node,
+            const std::set<ActionType> &                                controller_actions,
+            const std::set<ActionType> &                                environment_actions,
+            std::set<SearchTreeNode<Location, ActionType, ConstraintSymbolType> *> *visited)
```

Comment:
Why a pointer to the set instead of a reference?

Reply:
Yes, to make it clear that this is a single set that is passed around (rather than a const parameter to the function).
Thus, this is an output parameter in some sense, where:
> Use non-const pointers to represent optional outputs and optional input/output parameters.

https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs

Reply:
OK, I followed your suggestion and changed it to non-const ref.